bin/aomp_common_vars

```diff
@@ -31,14 +31,25 @@ AOMP_VERSION_STRING=${AOMP_VERSION_STRING:-"$AOMP_VERSION-$AOMP_VERSION_MOD"}
 ROCM_EXPECTED_MODVERSION=${ROCM_EXPECTED_MODVERSION:-6.2.4}
 export AOMP_VERSION_STRING AOMP_VERSION AOMP_VERSION_MOD ROCM_VERSION ROCM_EXPECTED_MODVERSION
 
-# Set install directory and the link directory.
-# AOMP will be a symbolic link AOMP_INSTALL_DIR is the versioned dir name
-AOMP=${AOMP:-$HOME/rocm/aomp}
-AOMP_INSTALL_DIR=${AOMP}_${AOMP_VERSION_STRING}
-
-# For example, if AOMP_REPOS=$HOME/git/aomp11, then the primary aomp repo
-# is stored in $HOME/git/aomp11/aomp and the mono repo is stored in
-# $HOME/aomp/llvm-project.
+# Determine AOMP_INSTALL_DIR based on AOMP and if AOMP ends in lib/llvm
+# strip off the lib/llvm. BUILD scripts should only use AOMP_INSTALL_DIR, not AOMP.
+
+# NOTE regarding test scripts and test makefiles:
+# TEST scripts or Makefiles should use AOMP env variable to find the compiler
+# and runtimes. AOMP has a historic default to $HOME/rocm/aomp. But with the
+# new convention, that follows the ROCm directory structure, AOMP should really
+# be set to $HOME/rocm/aomp/lib/llvm. Some test scripts detect this historic
+# default and add the lib/llvm to find the compiler. Since AOMP uses the same
+# directory structure as /opt/rocm, setting AOMP=/opt/rocm/lib/llvm will use the
+# ROCm compiler.
+
+AOMP=${AOMP:-$HOME/rocm/aomp} # Assume test scripts will correct this.
```

Comment:
Most team members were presetting AOMP to end in lib/llvm.   This strips that off to set AOMP_INSTALL_DIR.   If you were using the default listed here, then nothing is stripped and build progresses correctly.  However, since test scripts assume the compiler (e.g. clang++) is located at $AOMP (e.g. $AOMP/bin/clang++) , those test scripts will fail with this default value of AOMP.   So a lot of people were presetting AOMP=/work/$USER/rocm/aomp/lib/llvm.    Which causes build_aomp.sh to fail.   This PR strips off lib/llvm for setting AOMP_INSTALL_DIR and does not modify AOMP.   So now one can build if AOMP is set to something/lib/llvm.